src/components/features/Layout/ProductDetail.tsx

```diff
@@ -0,0 +1,110 @@
+import { AddIcon, MinusIcon } from '@chakra-ui/icons';
+import { Box, Button, Flex, IconButton, Image, Text } from '@chakra-ui/react';
+import { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import { useAuth } from '@/provider/Auth';
+import { RouterPath } from '@/routes/path';
+
+import useProductDetail from '../../../api/hooks/useProductDetail';
+
+interface ProductDetailProps {
+  productId: string;
+}
+
+const ProductDetail: React.FC<ProductDetailProps> = ({ productId }) => {
+  const productDetail = useProductDetail(productId);
+  const [quantity, setQuantity] = useState<number>(1);
+  const [price, setPrice] = useState<number>(0);
+  const navigate = useNavigate();
+  const authInfo = useAuth();
+
+  useEffect(() => {
+    if (productDetail) {
+      setPrice(quantity * productDetail.detail.price.sellingPrice);
+    }
+  }, [productDetail, quantity]);
+
+  if (!productDetail) {
+    return <Text>로딩 중...</Text>;
+  }
+
+  const handleQuantityChange = (value: number) => {
+    if (value < 1 || value > 99) return;
+    setQuantity(value);
+    setPrice(value * productDetail.detail.price.sellingPrice);
+  };
+
+  const { detail } = productDetail;
+
+  const handleOrder = () => {
+    if (!authInfo) {
+      // 로그인하지 않은 경우 로그인 페이지로 이동
+      navigate(RouterPath.login);
+    } else {
+      // 로그인 되어 있는 경우 상품 주문페이지로 이동
+      navigate(RouterPath.order, { state: { productDetail: detail, quantity, price } });
```

Comment:
```suggestion
      const state: OrderLocationState = { productDetail: detail, quantity, price };
      navigate(RouterPath.order, { state });
```

location.state를 사용할 때는 state를 전달하는 곳과 소비하는 곳이 달라서 state의 변화가 있을 때 타입 안전하게 처리하기 위해서는 소비하는 곳에서 state의 타입을 선언하고 이를 전달하는 곳에서 선언한 타입에 의존하는게 좋습니다. 그럼 key가 다르게 전달되거나 타입이 다르다면 에러가 발생할테니까요.

추가로 생각해볼만한 지점으로는 `/order` URL에 직접 접근했을 때 어떻게 대응할지 문제가 있습니다. 이때는 location.state가 없을테니까요. location.state를 통해 전달하는 데이터 구조가 복잡하지 않다면 URL을 하나의 리소스로 바라보고 URL query string으로 전달하는 방법도 있습니다. 이렇게 하면 페이지에서 의존하고 있는 데이터가 URL에 존재하니까 유저가 URL을 통해 특정 페이지부터 다시 시작할 수 있다는 장점도 있구요. 선택의 문제이긴 합니다.

Reply:
오 /order URL 직접 접근 관련해서도 한 번 생각해보겠습니다 감사합니다.